### PR TITLE
EVG-6240 add support for subscriptions, refactor revision

### PR DESCRIPTION
**Pull request by ablack12**
We already had some functionality for subscriptions, but to avoid lots of repeated code there's some refactoring. Also moved revision from a query parameter to a field in the APIProjectRef.

### CODE REVIEW

File: model/event/subscriptions.go [hunk 1]
```diff
@@ -246,6 +246,30 @@ func RemoveSubscription(id string) error {
 	})
 }
 
+func IsSubscriptionAllowed(sub Subscription) (bool, string) {
+	for _, selector := range sub.Selectors {
+		if selector.Type == "object" {
+			if selector.Data == "build" || selector.Data == "version" || selector.Data == "task" {
+				if sub.Subscriber.Type == "jira-issue" || sub.Subscriber.Type == "evergreen-webhook" {
```
Comment (tychoish) [inline model/event/subscriptions.go]: are there constants for these types?

File: rest/data/subscription.go [hunk 2]
```diff
@@ -45,8 +114,34 @@ func (dc *DBSubscriptionConnector) GetSubscriptions(owner string, ownerType even
...
+				Message:    err.Error(),
+			}
+		}
+		if subscription == nil {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusNotFound,
+				Message:    "Subscription not found",
+			}
+		}
+		if subscription.Owner != owner {
```
Comment (bsamek) [inline rest/data/subscription.go]: How does this work with subscriptions at the project level? Is there a concept of an owner there?

Comment (ablack12) [reply rest/data/subscription.go]: At the project level, the owner is the project ID (I also verified this by checking some documents in staging).

File: model/event/subscriptions.go (hunk 1 above)
Comment (ablack12) [reply model/event/subscriptions.go]: Wow there are so many constants in triggers that should be in subscriptions.

**Pull request merged**